From 0ed3901e55222d5541474dc58e23cfbce003fba3 Mon Sep 17 00:00:00 2001
From: Davor Joja <davorjoja@logicbricks.com>
Date: Mon, 23 Jun 2014 19:58:32 +0200
Subject: [PATCH 276/290] Documentation: devicetree: bindings: drm: xylon:
 Updated logicvc binding

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Changed logicvc devicetree binding layer type and layer transparency
parameters from integer to string value.
Added background layer type parameter.
Added H-V sync and pixel data control bool parameters, needed when LCD
display is used.
Updated example.

Signed-off-by: Davor Joja <davorjoja@logicbricks.com>
(cherry picked from commit cc4ff590f0d380bb24f5701c09d0ab15013dd8e4)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../devicetree/bindings/drm/xylon/logicvc.txt      | 52 ++++++++++++++--------
 1 file changed, 33 insertions(+), 19 deletions(-)

diff --git a/Documentation/devicetree/bindings/drm/xylon/logicvc.txt b/Documentation/devicetree/bindings/drm/xylon/logicvc.txt
index da8f1af..9212bf0 100644
--- a/Documentation/devicetree/bindings/drm/xylon/logicvc.txt
+++ b/Documentation/devicetree/bindings/drm/xylon/logicvc.txt
@@ -1,27 +1,23 @@
 Binding for Xylon configurable video controller logiCVC IP core
 
 Required properties:
- - compatible: value must be "xylon,logicvc-4.00.a"
+ - compatible: "xylon,logicvc-4.00.a", "xylon,logicvc-4.01.a"
  - reg: MMIO base address and size of the logiCVC IP core address space
  - interrupts-parent: the phandle for interrupt controller
  - interrupts: the interrupt number
- - colorspace: video output colorspace
-      (0=RGB, 1=YUV 4:2:2, 2=YUV 4:4:4)
- - interface: video output interface
-      (0=Parallel, 1=ITU656)
  - pixel-stride: layer width in pixels
       Common for all available logiCVC standard layers.
 
  - layer_N: layer has its own configuration described with below properties,
       where N is layer ID in range 0 - 4
    - bits-per-pixel: layer bits per pixel configuration (16, 32)
-         Layer is configured to be used with specific pixel width in bits.
+         Layer is configured to be used with specified pixel width in bits.
          Pixels written to layer video memory must match in size to configured
          bits per pixel value.
-   - format: layer format (0=RGB, 1=YUV)
-         Layer is configured to be used with specific color format.
+   - type: layer type (rgb, yuv)
+         Layer is configured to be used with specified color space.
          Pixels written to layer video memory must match specified color format.
-   - transparency: layer transparency (0=Layer, 1=Pixel)
+   - transparency: layer transparency (layer, pixel)
          logiCVC layer can be configured to have transparency control on
          layer or pixel level.
          "Layer" mode enables controlling of layer transparency by changing
@@ -36,6 +32,23 @@ Optional properties:
       bits per pixel.
       If 16 or 32, last available layer is logiCVC background layer,
       with only specified bits per pixel value.
+ - background-layer-type: background layer type (rgb, yuv)
+      Must be used only when "background-layer-bits-per-pixel" exist.
+      If set to "rgb", in case of 32 bits per pixel, background color register
+      must be written with XRGB8888 value. In case of 16 bits per pixel,
+      background color register must be written with RGB565 value.
+      If set to "yuv", background color register must be written with XYUV8888
+      value.
+ - is-hsync-active-low: horizontal synchronization pulse polarity
+      If omitted, generated horizontal synchronization pulse polarity is logic
+      high "H".
+ - is-vsync-active-low: vertical synchronization pulse polarity
+      If omitted, generated vertical synchronization pulse polarity is logic
+      high "H".
+ - is-pixel-data-invert: output pixel data polarity
+      If omitted, logiCVC outputs pixel data at default interface polarity.
+ - is-pixel-data-output-trigger-high: output pixel data trigger polarity
+      If omitted, logiCVC outputs pixel data on falling edge of pixel clock.
  - is-readable-regs: all logiCVC registers are available for reading
       If omitted, only Interrupt Status, Power Control and IP Version registers
       are available for reading.
@@ -59,33 +72,34 @@ Example:
 		interrupt-parent = <&gic_0>;
 		interrupts = <0 59 4>;
 		background-layer-bits-per-pixel = <32>;
-		colorspace = <1>;
-		interface = <0>;
+		background-layer-type = "rgb";
+		is-hsync-active-low;
+		is-vsync-active-low;
 		is-readable-regs;
 		is-size-position;
 		pixel-stride = <2048>;
 		layer_0 {
 			address = <0x30000000>;
 			bits-per-pixel = <16>;
-			format = <0>;
-			transparency = <0>;
+			type = "rgb";
+			transparency = "layer";
 		};
 		layer_1 {
 			address = <0x30500000>;
 			bits-per-pixel = <32>;
-			format = <0>;
-			transparency = <0>;
+			type = "rgb";
+			transparency = "layer";
 		};
 		layer_2 {
 			address = <0x30E00000>;
 			bits-per-pixel = <32>;
-			format = <0>;
-			transparency = <1>;
+			type = "rgb";
+			transparency = "pixel";
 		};
 		layer_3 {
 			address = <0x31700000>;
 			bits-per-pixel = <16>;
-			format = <1>;
-			transparency = <0>;
+			type = "yuv";
+			transparency = "layer";
 		};
 	};
-- 
2.0.2

